src/translate_c/ast.zig

```diff
@@ -176,6 +178,7 @@ pub const Node = extern union {
         c_pointer,
         single_pointer,
         array_type,
+        null_sentinel_array_type,
```

Comment:
I'm thinking maybe it would be better to add a `null_sentinel` boolean to `Payload.Array` instead of introducing a new ast node type here? It would let me remove some ugly if statements and helper functions from `translate_c.zig`